xbmc/video/VideoDatabase.cpp

```diff
@@ -4919,11 +4927,69 @@ bool CVideoDatabase::GetArtForItem(int mediaId, const MediaType &mediaType, std:
     if (nullptr == m_pDS2)
       return false; // using dataset 2 as we're likely called in loops on dataset 1
 
-    std::string sql = PrepareSQL("SELECT type,url FROM art WHERE media_id=%i AND media_type='%s'", mediaId, mediaType.c_str());
+    std::string sql;
+    if (versionId != -1 && (mediaType == MediaTypeMovie || mediaType == MediaTypeVideoVersion))
+    {
+      int versionMediaId{-1};
+      if (versionId == VIDEO_VERSION_ID_ALL)
+      {
+        //! @todo get rid of "videos with versions as folder" hack!
+        sql = PrepareSQL("SELECT videoVersionIdFile FROM movie_view WHERE idMovie = %i AND "
+                         "isDefaultVersion = 1",
+                         mediaId);
+        m_pDS2->query(sql);
+        if (!m_pDS2->eof())
+          versionMediaId = m_pDS2->fv(0).get_asInt();
+      }
+      else if (mediaType == MediaTypeMovie)
+      {
+        // If we get a request for a movie, mediaId contains the movie id, and we additionally need
+        // the file id for below art table lookup, to fetch all art.
+        sql = PrepareSQL("SELECT idFile FROM videoversion WHERE idMedia = %i AND media_type = '%s' "
+                         "AND idType = %i",
+                         mediaId, MediaTypeMovie, versionId);
+        m_pDS2->query(sql);
+        if (!m_pDS2->eof())
+          versionMediaId = m_pDS2->fv(0).get_asInt();
+      }
+      else if (mediaType == MediaTypeVideoVersion)
+      {
+        // If we get a request for a videoversion, mediaId contains the file id, and we additionally
+        // need the movie id for below art table lookup, to fetch all art.
+        versionMediaId = mediaId;
+        sql = PrepareSQL("SELECT idMedia FROM videoversion WHERE idFile = %i AND media_type = '%s' "
+                         "AND idType = %i",
+                         mediaId, MediaTypeMovie, versionId);
+        m_pDS2->query(sql);
+        if (!m_pDS2->eof())
+          mediaId = m_pDS2->fv(0).get_asInt();
+      }
+      if (versionMediaId != mediaId)
```

Comment:
FYI in a database built by library export/import, the videoversion records have idFile = idMedia.
As a consequence this test is false and the sql remains one of the previous queries, which do not retrieve art, have the wrong number of colums and an exception is raised when reading the dataset. No need for you to rewrite at this time, I'll try to send you a patch rewriting the function.

Reply:
Thanks for the patch.